src/main/kotlin/org/jetbrains/research/ml/codetracker/server/TrackerQueryExecutor.kt

```diff
@@ -1,78 +1,132 @@
 import okhttp3.RequestBody.Companion.toRequestBody
-import okhttp3.Response
+import org.jetbrains.research.ml.codetracker.Plugin
+import org.jetbrains.research.ml.codetracker.models.Extension
 import org.jetbrains.research.ml.codetracker.tracking.StoredInfoWrapper
 import java.io.File
-import java.lang.IllegalStateException
+import java.io.PrintWriter
 import java.net.URL
 
+
 object TrackerQueryExecutor : QueryExecutor() {
 
     private val ACTIVITY_TRACKER_FILE = "ide-events${Extension.CSV.ext}"
     private const val CODE_TRACKER_FILE_FIELD = "codetracker"
     private const val ACTIVITY_TRACKER_FILE_FIELD = "activitytracker"
     private val activityTrackerPath = "${PathManager.getPluginsPath()}/activity-tracker/" + ACTIVITY_TRACKER_FILE
+    private const val DEFAULT_ACTIVITY_TRACKER_ID = "-1"
 
-    var activityTrackerKey: String? = null
+    var studentId: String? = null
 
     init {
-        StoredInfoWrapper.info.activityTrackerKey?.let {
-            activityTrackerKey = it
+        StoredInfoWrapper.info.studentId?.let {
+            studentId = it
         } ?: run {
-            initActivityTrackerInfo()
-            StoredInfoWrapper.updateStoredInfo(activityTrackerKey = activityTrackerKey)
+            initStudentId()
+            StoredInfoWrapper.updateStoredInfo(studentId = studentId)
         }
     }
 
-    private fun initActivityTrackerInfo() {
-        val currentUrl = URL(baseUrl + "activity-tracker-item")
-        logger.info("${Plugin.PLUGIN_ID}: ...generating activity tracker key")
+    private fun initStudentId() {
+        val currentUrl = URL(baseUrl + "student")
+        logger.info("${Plugin.PLUGIN_ID}: ...generating student id")
         val requestBody = ByteArray(0).toRequestBody(null, 0, 0)
         val request = Request.Builder().url(currentUrl).post(requestBody).build()
-        activityTrackerKey = executeQuery(request)?.let { it.body?.string() }
+        studentId = executeQuery(request)?.let { it.body?.string() }
     }
 
-    private fun sendTrackerData(file: File,
-                                fileFieldName: String,
-                                baseUrlSuffix: String,
-                                method: String,
-                                toAddActivityTrackerKey: Boolean = false,
-                                afterSuccessfulResponse: () -> Unit = { }) {
+    private fun getRequestForStudentQuery(
+        urlSuffix: String,
+        codeTrackerKey: String,
+        activityTrackerKey: String,
+        method: String = "PUT"
+    ): Request {
+        val json = "{\"diId\":$codeTrackerKey,\"atiId\":\"$activityTrackerKey\"}"
+        val body = json.toRequestBody("application/json".toMediaTypeOrNull())
+        return Request.Builder().url(baseUrl + urlSuffix).method(method, body).build()
+    }
+
+    private fun getRequestForSendingDataQuery(
+        urlSuffix: String,
+        file: File,
+        fileFieldName: String,
+        method: String = "POST"
```

Comment:
Please, remove the parameter 'method' and replace it in the method body with 'post' since it' never has the other value.